netlify/functions/manage-bot.ts

```diff
@@ -0,0 +1,363 @@
+import dedent from "dedent";
+import isEqual from "lodash/isEqual";
+import { createPullRequest } from "octokit-plugin-create-pull-request";
+import { File } from "octokit-plugin-create-pull-request/dist-types/types";
+const sharp = require("sharp");
+import { Response } from "@netlify/functions/src/function/response";
+import { Event } from "@netlify/functions/src/function/event";
+import { Octokit } from "@octokit/core";
+import { createOrUpdateTextFile } from "@octokit/plugin-create-or-update-text-file";
+import fetch from "node-fetch";
+
+import { token } from "../_config";
+import {
+  BotData,
+  BotDeleteRequest,
+  BotUpdateRequest,
+  PaymentAddress,
+} from "../../types/user";
+import { checkAuth, checkBotAuth } from "../util/auth-utils";
+import { getGroupEmails, sendConfirmationEmail } from "../util/user-utils";
+
+const OctokitClient = Octokit.plugin(createPullRequest, createOrUpdateTextFile);
+const octokit = new OctokitClient({ auth: token });
+
+async function editBot(
+  data: BotUpdateRequest,
+  username: string,
+  sessionToken: string
+): Promise<Response> {
+  const {
+    botName,
+    image,
+    crew,
+    polygonAddress,
+    ethereumAddress,
+    relegated,
+  }: BotUpdateRequest = data;
+
+  if (!botName) {
+    return {
+      statusCode: 422,
+      body: JSON.stringify({
+        error: "Bot name is required",
+      }),
+    };
+  }
+
+  if (!polygonAddress || !polygonAddress.newValue) {
+    return {
+      statusCode: 422,
+      body: JSON.stringify({
+        error: "Polygon address is required",
+      }),
+    };
+  }
+
+  if (!crew || !crew.newValue || crew.newValue.length < 1) {
+    return {
+      statusCode: 422,
+      body: JSON.stringify({
+        error: "Bot crew must have at least 1 member",
+      }),
+    };
+  }
+
+  let oldBotData: BotData;
```

Comment:
what I mean is that `let oldBotData` isn't any less "constant" than `const paymentAddresses` — **both** of those would simply be `let` if I were the one writing the rules! (And save `const` for things that are just more semantically "constants" rather than "variables that, as usual, happen to not need re-assigning".)

Anyway… moving on :-)

Reply:
Sure, let's discuss a code standard going forward.